src/test/ui/consts/const-eval/issue-52443.stderr

```diff
@@ -43,7 +43,28 @@ error[E0080]: could not evaluate repeat length
 LL |     [(); { for _ in 0usize.. {}; 0}]; //~ ERROR calls in constants are limited to constant functions
    |          ^^^^^^^^^^^--------^^^^^^^
    |                     |
-   |                     calling non-const fn `<I as std::iter::IntoIterator><std::ops::RangeFrom<usize>>::into_iter`
+   |                     inside call to `std::iter::range::<impl std::iter::Iterator for std::ops::RangeFrom<A>><usize>::next`
+   | 
+  ::: $SRC_DIR/libcore/ptr.rs:LL:COL
```

Comment:
Diagnostics improvement or regression. Depending on your view. We used to stop at the first non-const fn, now we just stop if/when we fail to actually evaluate. An error is reported above about non const fn calls and other non-const things happening here.

Reply:
We could probably make it a `delay_span_bug`, but i fear that we'll overlook some case and then run into that hypothetical ICE at some point.

These kind of errors only happen for repeat/array lengths and enum variant discriminant initializers (or constants used in such). Everything else reports only the other errors above.

Alternatively we can try to figure out a scheme that prevents const eval of constants with const qualification errors.